batchflow/batch.py

```diff
@@ -582,168 +505,87 @@ def apply_transform(self, func, *args, **kwargs):
+        # unbind the method to pass self explicitly
+        transform = parallel(type(self)._apply_once)
+        return transform(self, *args, func=func, p=p, **kwargs_full)
 
-    @action
-    def apply_transform_all(self, func, *args, src=None, dst=None, p=None, **kwargs):
-        """ Apply a function the whole batch at once
+    def _apply_once(self, item, *args, func=None, p=None, **kwargs):
+        """ Apply a function to each item in the batch.
 
         Parameters
         ----------
+        item
+            an item of component data (in accordance with init function)
+
         func : callable
             a function to apply to each item from the source
 
-        src : str or array
-            the source to get data from, can be:
-
-            - str - a component name, e.g. 'images' or 'masks'
-            - array-like - a numpy-array, list, etc
-
-        dst : str or array
-            the destination to put the result in, can be:
-
-            - None
-            - str - a component name, e.g. 'images' or 'masks'
-            - array-like - a numpy-array, list, etc
-
-        p : float or None
-            probability of applying transform to an element in the batch
-
-            if not None, indices of relevant batch elements will be passed ``func``
-            as a named arg ``indices``.
+        p : None or int
+            whether to apply func to an element in the batch (yes if None or 1)
 
         args, kwargs
             other parameters passed to ``func``
-
-        Notes
-        -----
-        apply_transform_all does the following::
-
-            self.dst = func(self.src, *args, **kwargs)
-
-        When ``p`` is passed, random indices are chosen first and then passed to ``func``::
-
-            self.dst = func(self.src, *args, indices=random_indices, **kwargs)
-
-        Examples
-        --------
-
-        ::
-
-            apply_transform_all(make_masks_fn, src='images', dst='masks')
-            apply_transform_all(MyBatch.make_masks, src='images', dst='masks')
-            apply_transform_all(custom_crop, src='images', dst='augmented_images', p=.2)
-
         """
-        if not isinstance(dst, str) and not isinstance(src, str):
-            raise TypeError("At least one of dst and src should be attribute names, not arrays")
-
-        if src is None:
-            _args = args
-        else:
-            if isinstance(src, str):
-                src_attr = getattr(self, src)
-            else:
-                src_attr = src
-            _args = tuple([src_attr, *args])
-
-        if p is not None:
-            indices = np.where(np.random.binomial(1, p, len(self)))[0]
-            kwargs['indices'] = indices
-        tr_res = func(*_args, **kwargs)
-
-        if dst is None:
-            pass
-        elif isinstance(dst, str):
-            setattr(self, dst, tr_res)
-        else:
-            dst[:] = tr_res
-        return self
+        if p is None or p == 1:
```

Comment:
Special treatment for named expression `P` in `inbatch_parallel` is very hard to anticipate

Reply:
I don't think I understand your comment